File: 2026/Company/Company.Web/Modules/Sales/AccountType/AccountTypeRow.cs
using Company.Administration.Entities;

namespace Company.Sales;

[ConnectionKey("Sales"), Module("Sales"), TableName("[account].[AccountType]")]
[DisplayName("Account Type"), InstanceName("Account Type")]
[ReadPermission(SalesPermissionKeys.Currency.View)]
[ModifyPermission(SalesPermissionKeys.Currency.Modify)]
[DeletePermission(SalesPermissionKeys.Currency.Delete)]
[ServiceLookupPermission("SalesPermission:General")]
[LookupScript("Sales.AccountType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class AccountTypeRow : LoggingRow<AccountTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Account Type Id"), Column("AccountTypeID"), Identity, IdProperty]
    public int? AccountTypeId { get => fields.AccountTypeId[this]; set => fields.AccountTypeId[this] = value; }

    [DisplayName("Description"), Size(1073741823), QuickSearch, NameProperty]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field AccountTypeId;
        public StringField Description;
        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}

File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/AccountTypeRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface AccountTypeRow {
    AccountTypeId?: number;
    Description?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class AccountTypeRow {
    static readonly idProperty = 'AccountTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Description';
    static readonly localTextPrefix = 'Sales.AccountType';
    static readonly lookupKey = 'Sales.AccountType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<AccountTypeRow>('Sales.AccountType') }
    static async getLookupAsync() { return getLookupAsync<AccountTypeRow>('Sales.AccountType') }

    static readonly deletePermission = 'Company:Currency:Delete';
    static readonly insertPermission = 'Company:Currency:Modify';
    static readonly readPermission = 'Company:Currency:View';
    static readonly updatePermission = 'Company:Currency:Modify';

    static readonly Fields = fieldsProxy<AccountTypeRow>();
}